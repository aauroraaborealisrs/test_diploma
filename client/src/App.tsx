import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import Header from "./components/Header";
import Login from "./components/Login";
import Register from "./components/Register";
import AssignAnalysis from "./components/trainers/AssignAnalysis";
import SubmitAnalysis from "./components/students/SubmitAnalysis";
import DisplayAnalysis from "./components/trainers/DisplayAnalysis";
import AssignedAnalyses from "./components/trainers/AssignedAnalyses";
import UserAnalyses from "./components/students/UserAnalyses";
import ProtectedRoute from "./components/ProtectedRoute"; // üî• –ù–æ–≤—ã–π –∫–æ–º–ø–æ–Ω–µ–Ω—Ç
import Page404 from "./components/Page404";
import UserDashboard from "./components/students/UserDashboard";
import ProfileForm from "./components/ProfileForm";
import SuccessModal from "./components/shared/SuccessModal";

const App: React.FC = () => {
  return (
    <Router>
      <div>
        <Header />
        <Routes>
          {/* –û—Ç–∫—Ä—ã—Ç—ã–µ –º–∞—Ä—à—Ä—É—Ç—ã */}
          <Route path="/register" element={<Register />} />
          <Route path="/login" element={<Login />} />
          <Route path="/not-found" element={<Page404 />} />

          {/* üîí –ú–∞—Ä—à—Ä—É—Ç—ã —Ç–æ–ª—å–∫–æ –¥–ª—è —Å—Ç—É–¥–µ–Ω—Ç–æ–≤ */}
          <Route element={<ProtectedRoute allowedRoles={["student"]} />}>
            <Route path="/submit-analysis/:assignment_id" element={<SubmitAnalysis />} />
            <Route path="/my-analysis" element={<UserAnalyses />} />
            <Route path="/dashboard" element={<UserDashboard />} />
            <Route path="/profile" element={<ProfileForm />} />

          </Route>

          {/* üîí –ú–∞—Ä—à—Ä—É—Ç—ã —Ç–æ–ª—å–∫–æ –¥–ª—è —Ç—Ä–µ–Ω–µ—Ä–æ–≤ */}
          <Route element={<ProtectedRoute allowedRoles={["trainer"]} />}>
            <Route path="/assign-analysis" element={<AssignAnalysis />} />
            <Route path="/assignments" element={<AssignedAnalyses />} />
            <Route path="/analysis-results" element={<DisplayAnalysis />} />
          </Route>
        </Routes>
      </div>
    </Router>
  );
};

export default App;
