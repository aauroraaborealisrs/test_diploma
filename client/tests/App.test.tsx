// src/App.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../src/App'; // Adjust the path as needed

// Мокаем Header и AppRoutes, чтобы тест был изолированным
jest.mock('../src/components/Header', () => () => <div data-testid="header">HeaderMock</div>);
jest.mock('../src/components/AppRoutes', () => () => <div data-testid="routes">RoutesMock</div>);

describe('App component', () => {
  it('renders Header and AppRoutes inside a Router', () => {
    render(<App />);

    // Проверяем, что наши заглушки отрисовались
    expect(screen.getByTestId('header')).toBeInTheDocument();
    expect(screen.getByTestId('routes')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
