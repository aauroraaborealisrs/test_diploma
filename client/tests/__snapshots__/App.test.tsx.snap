// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="header"
  >
    HeaderMock
  </div>
  <div
    data-testid="routes"
  >
    RoutesMock
  </div>
</DocumentFragment>
`;
